test/unit/org/apache/cassandra/cql3/validation/operations/CQLVectorTest.java

```diff
@@ -248,25 +250,155 @@ public void token()
+                           "AS '" +
+                           "if (x == null || y == null) return null;" +
+                           "List<Integer> l = new ArrayList<Integer>(x); " +
+                           "l.addAll(y); " +
+                           "return l;'");
+        assertRows(execute(format("SELECT %s(value, [3, 4]) FROM %%s", f)), row(vector(1, 2, 3, 4)));
+        assertRows(execute(format("SELECT %s([2, 3], value) FROM %%s", f)), row(vector(2, 3, 1, 2)));
+        assertRows(execute(format("SELECT %s(null, null) FROM %%s", f)), row((Vector<Integer>) null));
+
+        // Test wrong arguments on function call
+        assertInvalidThrowMessage("cannot be passed as argument 0 of function " + f,
+                                  InvalidRequestException.class,
+                                  format("SELECT %s((int) 0, [3, 4]) FROM %%s", f));
+        assertInvalidThrowMessage("cannot be passed as argument 1 of function " + f,
+                                  InvalidRequestException.class,
+                                  format("SELECT %s([1, 2], (int) 0) FROM %%s", f));
+        assertInvalidThrowMessage("Invalid number of arguments in call to function " + f,
+                                  InvalidRequestException.class,
+                                  format("SELECT %s([1, 2]) FROM %%s", f));
+        assertInvalidThrowMessage("Invalid number of arguments in call to function " + f,
+                                  InvalidRequestException.class,
+                                  format("SELECT %s([1, 2], [3, 4], [5, 6]) FROM %%s", f));
+        assertInvalidThrowMessage("Unable to create a vector selector of type vector<int, 2> from 3 elements",
+                                  InvalidRequestException.class,
+                                  format("SELECT %s([1, 2, 3], [4, 5, 6]) FROM %%s", f));
+
+        // Test wrong types on function creation
+        assertInvalidThrowMessage("vectors may only have positive dimentions; given 0",
+                                  InvalidRequestException.class,
+                                  "CREATE FUNCTION %s (x vector<int, 0>) " +
+                                  "CALLED ON NULL INPUT " +
+                                  "RETURNS vector<int, 2> " +
+                                  "LANGUAGE java " +
+                                  "AS 'return x;'");
+        assertInvalidThrowMessage("vectors may only have positive dimentions; given 0",
+                                  InvalidRequestException.class,
+                                  "CREATE FUNCTION %s (x vector<int, 2>) " +
+                                  "CALLED ON NULL INPUT " +
+                                  "RETURNS vector<int, 0> " +
+                                  "LANGUAGE java " +
+                                  "AS 'return x;'");
+
+        // function reading and writing a udt vector field
+        String udt = createType("CREATE TYPE %s (v vector<int,2>)");
```

Comment:
There's some cleanup error on the test if you run it locally, presumably related to this. It'd be good to cleanup in case we ever need to run the test repeatedly.

Reply:
I think it's good you fixed it for the time being :-)